.github/actions/on_host_tests/action.yaml

```diff
@@ -0,0 +1,45 @@
+name: On Host Tests
+description: Runs on-host tests.
+inputs:
+  test_artifacts_key:
+    description: "Artifact key used to store test artifacts."
+    required: true
+  results_dir:
+    description: "Path to directory where test results are saved."
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: Download Artifacts
+      uses: actions/download-artifact@v4
+      with:
+        name: ${{ inputs.test_artifacts_key }}
+    - name: Extract Artifacts
+      shell: bash
+      run: |
+        set -x
+        mkdir -p ${GITHUB_WORKSPACE}/unit_test
+        cd ${GITHUB_WORKSPACE}/unit_test
+        tar xvf ${GITHUB_WORKSPACE}/test_artifacts.tar.xz
+    - name: Run Tests
+      id: run-tests
+      shell: bash
+      env:
+        GTEST_OUTPUT_ARG: ${{ inputs.results_dir }}
+        XVFB_SERVER_ARGS: "-screen 0 1920x1080x24i +render +extension GLX -noreset"
+      run: |
+        set -x
+        echo "Running env..."
+        env
+        pwd
+        ls
+        ls ${GITHUB_WORKSPACE}
+        echo "Attempting to run tests..."
+
+        cd ${GITHUB_WORKSPACE}/unit_test
+        for test_binary in $(ls -d out/**/*tests); do
+          # TODO(b/372303096): Need dedicated xvfb runner.
+          echo "Running tests for suite: ${test_binary}"
+          /usr/bin/xvfb-run -a --server-args="${XVFB_SERVER_ARGS}" ./$test_binary --gtest_filter=`tr '\n' ':' < ${GITHUB_WORKSPACE}/src/cobalt/testing/failing_tests.filter` --gtest_output="xml:${RESULTS_DIR}/${test_binary}_testoutput.xml" || true
```

Comment:
You are using `tr` here to to take the text file and pass it to args - neat trick.

I'd recommend try a bit with `jq` - json query tool. Would allow making filter files json - easier to check for correct syntax and also otherwise easily machine readable.

E.g : `jq -r '.filters | join(":")' file.json

And filters.json could look like

```
{
  "filters": [
    "-foo", "-bar", "baz"
   ]
}
```

Reply:
I think from your current test what was missing is that you need to enclose the filters in double quotes for the shell, e.g

```
--gtest_filter="thefilterspec goes here"
```

Because otherwise characters like `-` and `*` will get interpreted as shell tokens, not filterspec